public/app/features/plugins/tests/plugin_loader.test.ts

```diff
@@ -55,14 +53,17 @@ describe('Load App', () => {
     const m = await SystemJS.import(modulePath);
     expect(m.plugin).toBe(app);
 
-    const loaded = await importAppPlugin(meta);
-    expect(loaded).toBe(app);
+    // Importing the app should initialise the meta
+    const importedApp = await importAppPlugin(meta);
+    expect(importedApp).toBe(app);
     expect(app.meta).toBe(meta);
-    expect(app.initWasCalled).toBeTruthy();
-    expect(app.calledTwice).toBeFalsy();
 
-    const again = await importAppPlugin(meta);
-    expect(again).toBe(app);
-    expect(app.calledTwice).toBeTruthy();
+    // Importing the same app again doesn't initialise it twice
+    const importedAppAgain = await importAppPlugin(meta);
+    expect(importedAppAgain).toBe(app);
+    expect(app.init).toHaveBeenCalledTimes(1);
```

Comment:
Am I wrong or the previous test was asserting the `init` method was called twice?

Reply:
Yeah, it's a tricky one. `app.calledTwice` was becoming true [in case the app plugin meta already got initialised](https://github.com/grafana/grafana/pull/86624/files#diff-d5eeb2b4949eb1aec48069a8cc8f6ab67d92013987fd83b3dfd8cdb16d96286fL24), and it was basically checking if the meta was already set before with the same value. Instead of that I was only checking here if the `init()` is not called again if the plugin was already imported (due to the cache), and that it returns the same app. What we could do is to also add a check for the `meta`?

```ts
expect(importedAppAgain.meta).toBe(meta);
```